test/TestProjects/SingleTopLevelClientWithOperations-LowLevel/Generated/TopLevelClientWithOperationClient.cs

```diff
@@ -0,0 +1,134 @@
+
+        /// <summary> Initializes a new instance of TopLevelClientWithOperationClient for mocking. </summary>
+        protected TopLevelClientWithOperationClient()
+        {
+        }
+
+        /// <summary> Initializes a new instance of TopLevelClientWithOperationClient. </summary>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <param name="endpoint"> server parameter. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="credential"/> is null. </exception>
+        public TopLevelClientWithOperationClient(AzureKeyCredential credential, Uri endpoint = null, TopLevelClientWithOperationClientOptions options = null)
+        {
+            if (credential == null)
+            {
+                throw new ArgumentNullException(nameof(credential));
+            }
+            endpoint ??= new Uri("http://localhost:3000");
+            options ??= new TopLevelClientWithOperationClientOptions();
+
+            _clientDiagnostics = new ClientDiagnostics(options);
+            _keyCredential = credential;
+            _pipeline = HttpPipelineBuilder.Build(options, new HttpPipelinePolicy[] { new LowLevelCallbackPolicy() }, new HttpPipelinePolicy[] { new AzureKeyCredentialPolicy(_keyCredential, AuthorizationHeader) }, new ResponseClassifier());
+            _endpoint = endpoint;
+        }
+
+        /// <param name="context"> The request context. </param>
+#pragma warning disable AZC0002
+        public virtual async Task<Response> OperationAsync(RequestContext context = null)
+#pragma warning restore AZC0002
+        {
+            using var scope = _clientDiagnostics.CreateScope("TopLevelClientWithOperationClient.Operation");
+            scope.Start();
+            try
+            {
+                using HttpMessage message = CreateOperationRequest();
+                return await _pipeline.ProcessMessageAsync(message, _clientDiagnostics, context).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <param name="context"> The request context. </param>
+#pragma warning disable AZC0002
+        public virtual Response Operation(RequestContext context = null)
+#pragma warning restore AZC0002
+        {
+            using var scope = _clientDiagnostics.CreateScope("TopLevelClientWithOperationClient.Operation");
+            scope.Start();
+            try
+            {
+                using HttpMessage message = CreateOperationRequest();
+                return _pipeline.ProcessMessage(message, _clientDiagnostics, context);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        private volatile Client1 _cachedClient1;
```

Comment:
Why volatile?

Reply:
AFAIK, clients are multithreaded.

Reply:
Thinking about it again, we probably should use `syncObj` pattern. let me fix it.

Reply:
Our clients are threadsafe.  What about using https://docs.microsoft.com/en-us/dotnet/api/system.threading.lazyinitializer.ensureinitialized?view=net-5.0?

Reply:
This is implementation: https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Private.CoreLib/src/System/Threading/LazyInitializer.cs

It is basically a wrapper around `Volatile.Read` and `Interlocked.CompareExchange`. We can generate those as well, it will save us some memory on syncObj. Calling methods from `LazyInitializer` directly would be an overhead - they require factory delegates that may produce closures.

Reply:
There is even an open issue for this: https://github.com/dotnet/runtime/issues/26255

Reply:
Fixed